src/custom/utils/metadata.ts

```diff
@@ -43,3 +51,18 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
     },
   }
 }
+
+export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
+  try {
+    const { create } = await import('ipfs-http-client')
+    const client = create({ url: getIpfsUri() })
+    const doc = safeStringify.stableStringify(appDataDoc)
+    const { cid } = await client.add(doc)
+    await pinata.pinByHash(cid.toString())
+    const { digest } = multihashes.decode(new CID(cid.toString()).multihash)
```

Comment:
I would love to have these laso exposed in `api/ipfs` and have them tested in unit testing

Reply:
It cannot be tested yet because of this [issue](https://github.com/facebook/jest/issues/9983)